src/main/java/com/softserveinc/ita/rozetka/page_objects/OrderPage.java

```diff
@@ -85,4 +85,40 @@ public ProductPage openProductPage(int productNumber) {
         switchTo().window(1);
         return new ProductPage();
     }
+
+    @Step("Order page: set Delivery city {city} ")
+    public OrderPage setDeliveryCity(String city) {
+        $x("//button[contains(@class, 'button--link')]").click();
+        $x("//div[@class='form__row js-city']//li[1]").click();
+        return this;
+    }
+
+    public String getDeliveryRegion(String deliveryCity) {
+        return $x("//span[contains(@class,'deliveries-heading')]").text();
+    }
+
+    @Step("Order page: open delivery City page")
+    public DeliveryCityPage openDeliveryCityPage() {
+        $x("//button[contains(@class, 'button--link')]").click();
+        return new DeliveryCityPage();
+    }
+
+    @Step("Order page: set delivery option {deliveryOption}")
+    public OrderPage setDeliveryOption(DeliveryOption deliveryOption) {
+        $x(format("(//fieldset[contains(@class, 'checkout-block')]//ul)[4]/li//span[contains(text(), '%s')]", deliveryOption.getPartialLocatorDeliveryOption())).click();
+        return this;
+    }
+
+    @Step("Order page: set delivery point by address")
+    public String setDeliveryPointByAddress(String Street, String houseNbr) {
+        $x("//button[contains(@class, 'dropdown-button')]").click();
+        $x("//input[@id='searchPickupDelivery']").setValue(Street);
+        $x(format("//div[contains(@class, 'autocomplete__item') and contains(text(), '%s')]", houseNbr)).click();
+        return $x("//button[contains(@class, 'dropdown-button')]").text();
+    }
+
+    public String getDeliveryPointTimesheet() {
+        return $x("//dl[contains(@class, 'schedule')]").text();
```

Comment:
this doesn't have waits in it, will this be enough for the test to pass?

Reply:
it worked in all tests (besides, the value is read at once - not a list)